### ISSUE TITLE
use-package-always-defer duplicates some binding autoloads

### ISSUE THREAD

**Issue by fice-t** [jwiegley/use-package#341]
Consider two use-package forms:

```elisp
(use-package test
  :bind (("C-c" . flargle)))

(use-package example
  :bind (("Moo" . bagel)))
```

After evaluating the first form, with `use-package-always-defer` set to `nil`, the second form expands in part to:

```elisp
(progn (unless (fboundp (quote bagel)) (autoload (function bagel) "example" nil t)) (ignore (let* (... ... ... ...) (add-to-list ... ...) (define-key ... key ...))))
```

while with `use-package-always-defer` set to `t`, it expands partially to:

```elisp
(progn (unless (fboundp (quote bagel)) (autoload (function bagel) "example" nil t)) (unless (fboundp (quote flargle)) (autoload (function flargle) "example" nil t)) (ignore (let* (... ... ... ...) (add-to-list ... ...) (define-key ... key ...))))
```
The issue here is the `(unless (fboundp (quote flargle)) (autoload (function flargle) "example" nil t))` which I believe shouldn't be there.

Of course the extra autoload doesn't actually run due to the `fboundp` check, but I don't see why the duplicated block is there at all.

This doesn't seem to happen when both forms have `:defer t` in them, which contradicts the documentation of `use-package-always-defer`.

**Comment by npostavs** [comment]
It's due to modifying a quoted list, easy to fix: see #342.

**Comment by fice-t** [comment]
Ah, that fixes it. Thanks.
